src/LazyOperations/InverseLinearMap.jl

```diff
@@ -0,0 +1,333 @@
+a single `InverseLinearMap` instance.
+Again we can make use of the conversion for convenience.
+
+```jldoctest constructors
+julia> B = transpose(A); ilm2 = InverseLinearMap(B, ilm)
+InverseLinearMap{Int64,BallInf{Int64,Array{Int64,1}},Int64,Array{Int64,2}}([14 11 11; 11 14 11; 11 11 14], BallInf{Int64,Array{Int64,1}}([0, 0, 0], 1))
+
+julia> ilm2.M == B*A
+true
+```
+
+The application of a `InverseLinearMap` to a `ZeroSet` or an `EmptySet` is
+simplified automatically.
+
+```jldoctest constructors
+julia> InverseLinearMap(A, ZeroSet{Int}(3))
+ZeroSet{Int64}(3)
+```
+"""
+struct InverseLinearMap{N<:Real, S<:LazySet{N},
+                 NM, MAT<:AbstractMatrix{NM}} <: AbstractAffineMap{N, S}
+    M::MAT
+    X::S
+
+    # default constructor with dimension match check
+    function InverseLinearMap(M::MAT, X::S;
+                                check_invertibility::Bool=true) where {N<:Real,
+                                    S<:LazySet{N}, NM, MAT<:AbstractMatrix{NM}}
+        @assert dim(X) == size(M, 1) "a linear map of size $(size(M)) cannot " *
+            "be applied to a set of dimension $(dim(X))"
+        if check_invertibility
+            @assert isinvertible(M) "the linear map is not invertible"
+        end
+        return new{N, S, NM, MAT}(M, X)
+    end
+end
+
+
+# convenience constructor from a UniformScaling
+function InverseLinearMap(M::UniformScaling{N}, X::LazySet) where {N}
+    if M.λ == one(N)
+        return X
+    end
+    return InverseLinearMap(Diagonal(fill(M.λ, dim(X))), X)
+end
+
+
+# convenience constructor from a scalar
+function InverseLinearMap(α::Real, X::LazySet)
+    n = dim(X)
+    return InverseLinearMap(sparse(α * I, n, n), X)
+end
+
+# combine two linear maps into a single linear map
+function InverseLinearMap(M::AbstractMatrix, ilm::InverseLinearMap)
+    return InverseLinearMap(ilm.M * M, ilm.X)
+end
+
+# ZeroSet is "almost absorbing" for InverseLinearMap (only the dimension changes)
+function InverseLinearMap(M::AbstractMatrix{N}, Z::ZeroSet{N}) where {N}
+    @assert dim(Z) == size(M, 2) "a linear map of size $(size(M)) cannot " *
+            "be applied to a set of dimension $(dim(Z))"
+    return ZeroSet{N}(size(M, 1))
+end
+
+# EmptySet is absorbing for LinearMap
+function InverseLinearMap(M::AbstractMatrix, ∅::EmptySet)
+    return ∅
+end
+
+
+# --- AbstractAffineMap interface functions ---
+
+function matrix(ilm::InverseLinearMap)
+    return ilm.M
+end
+
+function vector(ilm::InverseLinearMap{N}) where {N}
+    return spzeros(N, dim(ilm))
+end
+
+function set(ilm::InverseLinearMap)
+    return ilm.X
+end
+
+"""
+    dim(ilm::InverseLinearMap)
+
+Return the dimension of an inverse linear map.
+
+### Input
+
+- `ilm` -- inverse linear map
+
+### Output
+
+The ambient dimension of the inverse linear map.
+"""
+function dim(ilm::InverseLinearMap)
+    return size(ilm.M, 1)
+end
+
+"""
+    σ(d::AbstractVector{N}, ilm::InverseLinearMap{N}) where {N<:Real}
```

Comment:
```suggestion
    σ(d::AbstractVector{N}, ilm::InverseLinearMap{N}) where {N}
```
it is not necessary to annotate `N<:Real`. 

we can also remove `N` altogether, here or in anther PR.

please note that the comment about removing `N<:Real` applies to all other occurrences in this file.

Reply:
If I change the signature from
```
function σ(d::AbstractVector{N}, ilm::InverseLinearMap{N,S,NM,MAT}) where {N<:Real, S<:LazySet{N}, NM, MAT}
```
to
```
function σ(d::AbstractVector{N}, ilm::InverseLinearMap{N,S,NM,MAT}) where {N, S<:LazySet{N}, NM, MAT}
```
I got an error of ambiguity with another methods:

```
MethodError: σ(::Array{Float64,1}, ::InverseLinearMap{Float64,BallInf{Float64,Array{Float64,1}},Float64,Array{Float64,2}}) is ambiguous. Candidates:
  σ(d::AbstractArray{N,1}, am::AbstractAffineMap{N,S} where S<:LazySet{N}) where N<:Real in LazySets at C:\Users\danie\.julia\dev\LazySets\src\Interfaces\AbstractAffineMap.jl:71
  σ(d::AbstractArray{N,1}, ilm::InverseLinearMap{N,S,NM,MAT}) where {N, S<:LazySet{N}, NM, MAT} in LazySets at C:\Users\danie\.julia\dev\LazySets\src\LazyOperations\InverseLinearMap.jl:163
Possible fix, define
  σ(::AbstractArray{N,1}, ::InverseLinearMap{N,S,NM,MAT}) where {N<:Real, S<:LazySet{N}, NM, MAT}

Stacktrace:
 [1] top-level scope at show.jl:641
 [2] include_string(::Function, ::Module, ::String, ::String) at .\loading.jl:1091
```

Suggestion?

Reply:
ok, i see. then i suggest to keep it like it was, and revisit after we do https://github.com/JuliaReach/LazySets.jl/pull/2276